editor/scene_tree_dock.cpp

```diff
@@ -1408,6 +1414,82 @@ void SceneTreeDock::_tool_selected(int p_tool, bool p_confirm_override) {
 					}
 					undo_redo->add_do_method(node, "set_unique_name_in_owner", false);
 					undo_redo->add_undo_method(node, "set_unique_name_in_owner", true);
+					if (node->is_exposed_in_owner()) {
+						undo_redo->add_undo_method(node, "set_exposed_in_owner", true);
+					}
+				}
+				undo_redo->commit_action();
+			}
+		} break;
+		case TOOL_TOGGLE_SCENE_EXPOSED: {
+			// Enabling/disabling based on the same node based on which the checkbox in the menu is checked/unchecked.
+			List<Node *>::Element *first_selected = editor_selection->get_selected_node_list().front();
+			if (first_selected == nullptr) {
+				return;
+			}
+			if (first_selected->get() == EditorNode::get_singleton()->get_edited_scene()) {
+				// Exclude Root Node. It should never be unique name in its own scene!
+				editor_selection->remove_node(first_selected->get());
+				first_selected = editor_selection->get_selected_node_list().front();
+				if (first_selected == nullptr) {
+					return;
+				}
+			}
+			bool enabling = !first_selected->get()->is_exposed_in_owner();
+
+			List<Node *> full_selection = editor_selection->get_full_selected_node_list();
+			EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+
+			if (enabling) {
+				Vector<Node *> new_exposed_nodes;
+				Vector<StringName> new_unique_names;
+				Vector<StringName> cant_be_set_unique_names;
+
+				for (Node *node : full_selection) {
+					if (node->is_exposed_in_owner()) {
+						continue;
+					}
+					if (node->get_owner() != get_tree()->get_edited_scene_root()) {
+						continue;
+					}
+
+					StringName name = node->get_name();
+					Node *uniqueNode = get_tree()->get_edited_scene_root()->get_node_or_null(UNIQUE_NODE_PREFIX + String(name));
+					if (new_unique_names.has(name) || (uniqueNode != nullptr && uniqueNode != this && uniqueNode->is_exposed_in_owner())) {
+						cant_be_set_unique_names.push_back(name);
+					} else {
+						new_exposed_nodes.push_back(node);
+						new_unique_names.push_back(name);
+					}
+				}
+				if (new_exposed_nodes.size()) {
+					undo_redo->create_action("Enable Exposed Node(s)");
```

Comment:
```suggestion
					undo_redo->create_action(TTR("Enable Exposed Node(s)"));
```
Undo/redo should be translated